realm/realm-library/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -131,6 +134,8 @@ public void clear(String className) {
      * during the transaction {@link #cancelTransaction()} will be called instead of {@link #commitTransaction()}.
      *
      * @param transaction {@link io.realm.DynamicRealm.Transaction} to execute.
+     * @throws IllegalArgumentException if the {@code transaction} is {@code null}
+     * @throws RuntimeException if {@code transaction} could not be cancelled
```

Comment:
I would remove this line, as it doesn't really provide much info nor is it something users can guard against. It is also in our _now_ public Javadoc guidelines :) : https://github.com/realm/realm-java/blob/master/CONTRIBUTING.md#javadoc